webapp/src/lib/components/spot-component/spot-location-create-form.svelte

```diff
@@ -5,100 +5,280 @@
-        bind:selected={countryCode}
-    >
-        <SelectItem value="CA" text="Canada" />
-    </Select>
-    <TextInput
-        required
-        labelText="Postal code"
-        name="postal-code"
-        placeholder="Postal code"
-        readonly={isReadOnly}
-        bind:value={postalCode}
-    />
-    <p>Utilities</p>
-    <Checkbox
-        name="shelter"
-        style={isReadOnly ? 'pointer-events: none;' : ' '}
-        labelText="shelter"
-        bind:checked={hasShelter}
-    />
-    <Checkbox
-        name="plug-in"
-        labelText="Plug-in"
-        style={isReadOnly ? 'pointer-events: none;' : ' '}
-        bind:checked={hasPlugIn}
-    />
-    <Checkbox
-        name="charging-station"
-        labelText="Charging Station"
-        style={isReadOnly ? 'pointer-events: none;' : ' '}
-        bind:checked={hasChargingStation}
-    />
-    {#if !isReadOnly}
-        <Button type="submit">Submit</Button>
-    {/if}
-</Form>
+<!-- svelte-ignore a11y-click-events-have-key-events a11y-no-static-element-interactions-->
+<div on:click={handleClickOutside}>
+    <Form on:submit={handleSubmit}>
+        {#if !isReadOnly}
+            <Search
+                type="text"
+                bind:value={searchQuery}
+                placeholder="Enter an address... [At least 3 characters required]"
+                name="search"
+                on:input={handleInput}
+            />
+            {#if results?.length > 0 && dropdownOpen}
+                <ul class="dropdown">
+                    {#each results as result}
+                        <!-- svelte-ignore a11y-click-events-have-key-events a11y-no-noninteractive-element-interactions -->
+                        <li on:click={() => handleSelect(result)}>
+                            {result.properties.formatted}
+                        </li>
+                    {/each}
+                </ul>
+            {/if}
+        {/if}
+        <Modal
+            open={warningModalOpen}
+            modalHeading="Error - Please check Address"
+            primaryButtonText="Re-enter address"
+            on:close
+            on:open
+            on:submit={() => (warningModalOpen = false)}
+        >
+            {warningText}
+        </Modal>
+        <TextInput
+            required
+            name="street-address"
+            placeholder="House # & Street Address"
+            readonly={isReadOnly}
+            bind:value={streetAddress}
+            on:input={handleEdit}
+            on:change={handleEdit}
+        />
+        <TextInput
+            required
+            name="city"
+            placeholder="City"
+            readonly={isReadOnly}
+            bind:value={city}
+            on:change={handleEdit}
+        />
+        <TextInput
+            required
+            name="postal-code"
+            placeholder="Postal code"
+            readonly={isReadOnly}
+            bind:value={postalCode}
+            on:change={handleEdit}
+        />
```

Comment:
Can you add some labels to these field?

Reply:
Adding a labelText makes the form look a bit unformatted, without the labels just having placeholders look cleaner.